src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1127,6 +1127,26 @@ void primitive_inst::do_runtime_skip_permute() {
         set_can_be_optimized(false);
         return;
     }
+
+    auto check_dependency_all_cpu = [&]() -> bool {
+        for (auto& dep : dependencies()) {
+            if (!is_user_cpu(&(dep.first->get_node())))
+                return false;
+        }
+        return true;
+    };
+
+    auto needs_sync = _needs_completion_event;
+    for (auto u : get_user_insts()) {
+        needs_sync |= u->_needs_completion_event;
+    }
+
+    if (get_network().get_stream().get_queue_type() == QueueTypes::in_order && _needs_completion_event && !check_dependency_all_cpu()) {
+        GPU_DEBUG_TRACE_DETAIL << "--- Cannot optimize because in order queue, and" << id() << " needs syncronization" << std::endl;
+        set_can_be_optimized(false);
```

Comment:
If the problem is related to synchronization, then I think that instead of doing explicit permute we can just schedule a barrier and provide it's event.

Reply:
updated, will sync with @yeonbok for a final solution

Reply:
IMO, the issue is not only for permute but for nested runtime optimized kernels. 
So I think we should handle this in a more general way. 
Current thought is we need to add additional function to scan nested users whether they have (node->need_completion_event &&   node->is_runtime_skippable()) 
Maybe we'll have false positive by doing this & need to check perf impact.